crates/chain/src/keychain/txout_index.rs

```diff
@@ -199,7 +313,10 @@ impl<K: Clone + Ord + Debug> KeychainTxOutIndex<K> {
     ) -> impl DoubleEndedIterator<Item = (K, u32, OutPoint, &TxOut)> {
         self.inner
             .txouts_in_tx(txid)
-            .map(|((k, i), op, txo)| (k.clone(), *i, op, txo))
+            .filter_map(|((desc_id, i), op, txo)| {
+                self.keychain_of_descriptor_id(desc_id)
+                    .map(|k| (k.clone(), *i, op, txo))
+            })
     }
 
     /// Return the [`TxOut`] of `outpoint` if it has been indexed.
```

Comment:
We need to update the docs to say that we will only return if the descriptor has corresponding keychain id (somehow).